File: examples/python/list_received.py
"""List your invoices via the getpeppr API."""

import requests

BASE_URL = "https://api.getpeppr.dev"
API_KEY = "sk_sandbox_abc123..."

response = requests.get(
    f"{BASE_URL}/v1/invoices",
    headers={"Authorization": f"Bearer {API_KEY}"},
    params={"limit": 10, "offset": 0},
    timeout=30,
)

response.raise_for_status()
result = response.json()

print(f"Total invoices: {result['meta']['total']} (showing {len(result['data'])})")

for invoice in result["data"]:
    print(f"  {invoice['id']}: {invoice['number']} — {invoice['status']}")

File: examples/typescript/list-received.ts
import { Peppol } from "@getpeppr/sdk";

const peppol = new Peppol({ apiKey: "sk_sandbox_..." });

// List your invoices (paginated)
const page = await peppol.invoices.list({ limit: 10, offset: 0 });

console.log(`Total invoices: ${page.meta.totalCount} (showing ${page.data.length})`);

for (const invoice of page.data) {
  console.log(`  ${invoice.id}: ${invoice.number} — ${invoice.status}`);
}

// Paginate through all invoices
for await (const invoice of peppol.invoices.listAll()) {
  console.log(`${invoice.number}: ${invoice.status}`);
}
